addons/mail/static/src/messaging/component/activity/activity_tests.js

```diff
@@ -781,6 +781,103 @@ QUnit.test('activity click on edit', async function (assert) {
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (route === '/web/dataset/call_kw/mail.activity/activity_format') {
+                assert.step('activity_format');
+                return [{
+                    id: 12,
+                    can_write: true,
+                    icon: 'fa-check',
+                    res_id: 42,
+                    res_model: 'res.partner',
+                }];
+            }
+            return this._super(...arguments);
+        },
+        intercepts: {
+            do_action: function (ev) {
+                assert.step('do_action');
+                assert.strictEqual(
+                    ev.data.action.context.default_res_id,
+                    42,
+                    'Action should have the activity res id as default res id in context');
+                assert.strictEqual(
+                    ev.data.action.context.default_res_model,
+                    'res.partner',
+                    'Action should have the activity res model as default res model in context');
+                assert.strictEqual(
+                    ev.data.action.type,
+                    "ir.actions.act_window",
+                    'Action should be of type "ir.actions.act_window"');
+                assert.strictEqual(
+                    ev.data.action.res_model,
+                    "mail.activity",
+                    'Action should have "mail.activity" as res_model');
+                assert.strictEqual(
+                    ev.data.action.res_id,
+                    12,
+                    'Action should have activity id as res_id');
+                ev.data.options.on_close();
+            },
+        },
+    });
+    const activity = this.env.entities.Activity.create({
+        id: 12,
+        can_write: true,
+        icon: 'fa-times',
+        res_id: 42,
+        res_model: 'res.partner',
+    });
+    await this.createActivityComponent(activity);
+    assert.containsOnce(
+        document.body,
+        '.o_Activity',
+        "should have activity component"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Activity_editButton',
+        "should have activity edit button"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Activity_icon',
+        "should have activity icon"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Activity_icon .fa-times',
+        "should have initial activity icon"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_Activity_icon .fa-check',
+        "should not have new activity icon when not edited yet"
+    );
+
+    await afterNextRender(() => {
+        document.querySelector('.o_Activity_editButton').click();
+    });
+    assert.verifySteps(
+        ['do_action', 'activity_format'],
+        "should have call 'schedule activity' action correctly"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_Activity_icon .fa-times',
```

Comment:
Just asking, there is no other way to test it that doesn't depend on the fa icon?

Reply:
We could change activity title I suppose, the icon was the fastest way to write the test